### PR TITLE
Misc fixes: gcc warning and Coverity defects

### PR DESCRIPTION
**Pull request by ja2-derek**
Tried doing a build with gcc today, and saw a bunch of new compiler warnings. These should not affect anything.

### CODE REVIEW

**Comment by ja2-derek** [comment]
> 
> 
> [463dbdb](https://github.com/ja2-stracciatella/ja2-stracciatella/commit/463dbdbe15c792d9e34d92da34817f361dc007da) Hi ja2-derek, we did the same commit today.

Ah cool!

**Comment by ja2-derek** [comment]
Took away the signed-unsigned stuff. Thanks @wvx

**Comment by wvx** [comment]
> Took away the signed-unsigned stuff. Thanks @wvx

No problem if you prefer your commit. The commit was hidden, so it is understandable that we duplicated the effort and I was asked to separate the PR. Thanks for also looking at the compiler warnings.

**Comment by lynxlynxlynx** [comment]
So can the other commit be cherrypick it into your branch? Seems it does quite some more.

**Comment by ja2-derek** [comment]
> 
> 
> So can the other commit be cherrypick it into your branch? Seems it does quite some more.

Done!

File: src/game/Strategic/Strategic_AI.cc [hunk 1]
```diff
@@ -3139,7 +3140,7 @@ static void RequestHighPriorityGarrisonReinforcements(size_t iGarrisonID, UINT8
 	}
 	ubDstSectorX = (UINT8)SECTORX( gGarrisonGroup[ iGarrisonID ].ubSectorID );
 	ubDstSectorY = (UINT8)SECTORY( gGarrisonGroup[ iGarrisonID ].ubSectorID );
-	if( iBestIndex != -1 )
+	if( iBestIndex != (size_t)-1 )
```
Comment (ja2-derek) [inline src/game/Strategic/Strategic_AI.cc:3143]: How does it work? Won't the cast change `-1` into `INT_MAX`? Can it still compare to a signed int?

Comment (wvx) [reply src/game/Strategic/Strategic_AI.cc:3143]: Yes, as I understand it it should be something like SIZE_MAX which is OK if "size_t iBestIndex = -1;" is equivalent to "size_t iBestIndex = SIZE_MAX;" There are other cases where -1 was inteded to returned as SIZE_MAX (but specified only as -1) to flag a special case such as in "size_t const patrol_id = FindPatrolGroupIndexForGroupID(g.ubGroupID);"

Maybe, as you suggest, we should be using the limits, such as SIZE_MAX in this case, to be clear on the intention.

Comment (ja2-derek) [reply src/game/Strategic/Strategic_AI.cc:3143]: Aha. I didn't realize `iBestIndex` is unsigned

**Comment by lynxlynxlynx** [comment]
Thanks guys!

**Pull request merged**